packages/@lwc/integration-karma/test/events/memoization/index.spec.js

```diff
@@ -46,4 +47,35 @@ describe('deep listener', () => {
         elm.shadowRoot.querySelector('button').click();
         expect(window.clickBuffer).toEqual([1, 2]);
     });
+
+    // In this case, the click listener is re-bound on every render, because the referenced
+    // listener is scoped inside a <template for:each>. However, the focus and blur listeners
+    // are not re-bound because they are scoped to the component, so the event listener memoization optimization causes
+    // the listener to be bound once and never redefined.
+    it('does redefine the onClick for a list of deep click listeners', async () => {
+        const elm = createElement('x-mixed', { is: Mixed });
+        document.body.appendChild(elm);
+
+        elm.loggers = [{ id: 1, onClick: () => window.clickBuffer.push(1) }];
+        elm.mainLogger = {
+            onFocus: () => window.clickBuffer.push('focus'),
+            onBlur: () => window.clickBuffer.push('blur'),
+        };
+        await Promise.resolve();
+        elm.shadowRoot.querySelector('button').click();
+        elm.shadowRoot.querySelector('button').focus();
+        elm.shadowRoot.querySelector('button').blur();
+        expect(window.clickBuffer).toEqual([1, 'focus', 'blur']);
+
+        elm.loggers = [{ id: 2, onClick: () => window.clickBuffer.push(2) }];
+        elm.mainLogger = {
+            onFocus: () => window.clickBuffer.push('focus2'),
+            onBlur: () => window.clickBuffer.push('blur2'),
+        };
+        await Promise.resolve();
+        elm.shadowRoot.querySelector('button').click();
+        elm.shadowRoot.querySelector('button').focus();
+        elm.shadowRoot.querySelector('button').blur();
+        expect(window.clickBuffer).toEqual([1, 'focus', 'blur', 2, 'focus', 'blur']);
+    });
```

Comment:
@nolanlawson here's the test for memoization with mixed scope handlers. This is the intended behavior right?